### ISSUE TITLE
DataSync silently override createdAt/updatedAt custom fields

### ISSUE THREAD

**Issue by machi1990** [aerogear/graphback#1597]
If I have a model with custom fields 

```
"""
Represents action that is assigned to volunteer

@model
@versioned
"""
type Entity {
  id: ID!
  createdAt: DateTime
  completedAt: DateTime
  updatedAt: DateTime
}
```

When I start the server with the datasync plugin the two fields `createdAt`, `updatedAt` are silently overwritten and their type is changed to `String`. 

```
"""
Represents action that is assigned to volunteer

@model
@versioned
"""
type Entity {
  id: ID!
  createdAt: String
  completedAt: DateTime
  updatedAt: String
}
```

/cc @ssd71 @wtrocki @craicoverflow

**Comment by ssd71** [comment]
This is a sort of workaround that we decided until we add Graphback native DateTime scalars, current implementation is quite unstable(hacky) and is bound to change, it's generally not recommended to explicitly specify those fields as of now :smile:, I think we should decide something about this soon, before something else breaks

Practically speaking as far as I am aware of, it's not recommended to use any type other than string (or Double) for these fields because of the way this hack works.

**Comment by craicoverflow** [comment]
I see three approach options:

1. Detect if these fields exist and don't overwrite the fields. The issue with this is how does Graphback decide whether the scalar added by the user is a valid one. 

If a field `createdAt: Date` is specified by the user it is probably invalid and will not work for DataSync, since it does not have enough information about the time.

2. Overwrite the field as it is doing, because this is one of the main functions of the `@versioned` field and a caveat of adding it is Graphback will need to control this field.

3. Throw an error and don't start the server until the user removes their custom field.

My personal preference is approach 3, since a rule to follow is to not overwrite user specified code/configuration. I see approach 1 as causing issues as Graphback does not know until query execution if the scalar is valid or not.

**Comment by craicoverflow** [comment]
^^ Posted hours ago, GitHub would not let me comment.

**Comment by machi1990** [comment]
> I see three approach options:
> 
> 1. Detect if these fields exist and don't overwrite the fields. The issue with this is how does Graphback decide whether the scalar added by the user is a valid one.
> 
> If a field `createdAt: Date` is specified by the user it is probably invalid and will not work for DataSync, since it does not have enough information about the time.
> 
> 1. Overwrite the field as it is doing, because this is one of the main functions of the `@versioned` field and a caveat of adding it is Graphback will need to control this field.
> 2. Throw an error and don't start the server until the user removes their custom field.
> 
> My personal preference is approach 3, since a rule to follow is to not overwrite user specified code/configuration. I see approach 1 as causing issues as Graphback does not know until query execution if the scalar is valid or not.

Only landing to this issue now.

What was our conclusion about this? 

I think going with throwing an error (indicative) enough should be fine. This way, the user could either remove the `@versioned` annotation from the model or change the name of the impacted fields. 

/cc @ssd71 @craicoverflow to get your opinion on this.

**Comment by wtrocki** [comment]
@ssd71 can confirm, but we decided to move to `_deleted` etc. Looks it can be closed.

**Comment by ssd71** [comment]
@wtrocki The issue is about `@versioned` now, instead of `@datasync`, I think

@machi1990 That sounds good to me, but I'm unsure if I am missing any essential context/perspective so let's also wait for @craicoverflow opinoin as well

**Comment by craicoverflow** [comment]
> I think going with throwing an error (indicative) enough should be fine. This way, the user could either remove the `@versioned` annotation from the model or change the name of the impacted fields.

Yep, this is the best thing to do. Overwriting the field without an error can leave the developer wondering why their application is not working, best to give them the information to do it safely and adapt their schema.
